doc/usecases/core/UC_keyset_delete.md

```diff
@@ -0,0 +1,28 @@
+# Use Case: Delete `Key` from `KeySet`
```

Comment:
Please take care to be consistent with our terminology doc/help/elektra-glossary.md

We currently call this operation `pop`. But I am open to change it (e.g. by using remove for both removing from the keyset and removing from KDB). Please also change doc/help/elektra-glossary.md

Reply:
Pop is a completely stupid name for what it does, because pop has a pretty specific meaning when it comes to data structures. If had to guess, I'd say at some point the pop operation removed the key that was added last, or maybe the last key in keyset order.

In any case, I'm open to renaming this use case to "Remove". But I don't think we need consistent terminology here. In fact, I think for use cases it makes a lot more sense to use _common terminology_ instead of introducing Elektra specific terms. That's also why I used terms like "associative array" and "prefix lookup". IMO the Elektra specific terminology should introduced on a decision level. In fact, deciding the terminology we will use is inherently part of a decision.